extensions/gdx-controllers/gdx-controllers-gwt/src/com/badlogic/gdx/controllers/gwt/GwtControllers.java

```diff
@@ -140,67 +150,102 @@ public void onGamepadConnected(int index) {
 	}
 
 	@Override
-	public void onGamepadDisconnected(int index) {
+	public void onGamepadDisconnected (int index) {
 		GwtController controller = controllerMap.remove(index);
-		if(controller != null) {
-			synchronized(eventQueue) {
+		if (controller != null) {
+			synchronized (eventQueue) {
 				GwtControllerEvent event = eventPool.obtain();
 				event.type = GwtControllerEvent.DISCONNECTED;
 				event.controller = controller;
 				eventQueue.add(event);
 			}
 		}
 	}
-	
+
 	@Override
-	public void onGamepadUpdated(int index) {
+	public void onGamepadUpdated (int index) {
 		Gamepad gamepad = Gamepad.getGamepad(index);
 		GwtController controller = controllerMap.get(index);
 		if (gamepad != null && controller != null) {
 			// Determine what changed
 			JsArrayNumber axes = gamepad.getAxes();
-			JsArrayNumber buttons = gamepad.getButtons();
-			synchronized(eventQueue) {
+			JsArray<GamepadButton> buttons = gamepad.getButtons();
+			synchronized (eventQueue) {
 				for (int i = 0, j = axes.length(); i < j; i++) {
 					float oldAxis = controller.getAxis(i);
-					float newAxis = (float) axes.get(i);
+					float newAxis = (float)axes.get(i);
 					if (oldAxis != newAxis) {
 						GwtControllerEvent event = eventPool.obtain();
 						event.type = GwtControllerEvent.AXIS;
 						event.controller = controller;
-						event.code = i;						
+						event.code = i;
 						event.amount = newAxis;
 						eventQueue.add(event);
 					}
-				}			
+				}
 				for (int i = 0, j = buttons.length(); i < j; i++) {
-					float oldButton = controller.getButtonAmount(i);
-					float newButton = (float) buttons.get(i);
-					if (oldButton != newButton) {
-						if ((oldButton < 0.5f && newButton < 0.5f) || (oldButton >= 0.5f && newButton >= 0.5f)) {
-							controller.buttons.put(i, newButton);
-							continue;
+					float newButton = (float)buttons.get(i).getValue();
+					if (controller.isStandardMapping() && i >= 12 && i <= 15) {
+						int direction = controller.pov;
+						if (newButton < 0.5f) {
+							if (i == 12)
```

Comment:
Sure thing... I'll replace them with proper constants and update the PR later today